pkg/controller/fileobserver/observer_polling.go

```diff
@@ -0,0 +1,124 @@
+package fileobserver
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type pollingObserver struct {
+	reactors map[string][]reactorFn
+	files    map[string]string
+
+	reactorsMutex sync.RWMutex
+}
+
+func (o *pollingObserver) AddReactor(reaction reactorFn, files ...string) Observer {
+	o.reactorsMutex.Lock()
+	defer o.reactorsMutex.Unlock()
+	for _, f := range files {
+		glog.V(3).Infof("Adding reactor for file %q", f)
+		var err error
+		o.files[f], err = calculateFileHash(f)
```

Comment:
nit: only if f is new

Reply:
@sttts why? even if you re-add you probably want to calc the hash straight away, no?

Reply:
then existing reactors lose events.